style_guides/js_style_guide.md

```diff
@@ -37,112 +36,145 @@ cheap syntactic pleasures.
 
 Try to limit your lines to 80 characters. If it feels right, you can go up to 120 characters.
 
-## Use single quotes
-
-Use single quotes, unless you are writing JSON.
+## Use `const` for variables
 
-*Right:*
+Your variable references should rarely be mutable, so use `const` for almost
+everything. If you absolutely *must* mutate a reference, use `let`.
 
 ```js
+// good
+const foo = 'bar';
+
+// if absolutely necessary, OK
+let foo;
+
+// bad
 var foo = 'bar';
 ```
 
-*Wrong:*
+## Use single quotes for fixed strings
 
-```js
-var foo = "bar";
-```
+Use single quotes, unless you are writing JSON.
 
-## Opening braces go on the same line
+```js
+// good
+const foo = 'bar';
 
-Your opening braces go on the same line as the statement.
+// bad
+const foo = "bar";
+```
 
-*Right:*
+## Use template strings to interpolate variables into strings
 
 ```js
-if (true) {
-  console.log('winning');
-}
+// good
+const foo = `Hello, ${name}`;
+
+// bad
+const foo = 'Hello, ' + name;
 ```
 
-*Wrong:*
+## Use template strings to avoid escaping single quotes
+
+Because readability is paramount.
 
 ```js
-if (true)
-{
-  console.log('losing');
-}
-```
+// good
+const foo = `You won't believe this.`;
 
-Also, notice the use of whitespace before and after the condition statement.
+// bad
+const foo = 'You won\'t believe this.';
+```
 
-## Always use braces for multi-line code
+## Use object destructuring
 
-*Right:*
+This helps avoid temporary references and helps prevent typo-related bugs.
 
 ```js
-if (err) {
-  return cb(err);
+// best
+function fullName({ first, last }) {
```

Comment:
Use your best judgment, which is why both are listed as acceptable here.  You can still use `user` when it makes the function easier to understand, which is why the "good" example demonstrates that.  Personally, I'd start to question the amount of responsibilities of a function if that was important, but it's not something that's easy to draw a blanket statement about, which is why both are OK.

I think not being able to enforce something with a linting rule is all the more reason to include it in the styleguide.  I don't think we need a secondary "guide" for style.